### PR TITLE
feat: add doc for the greenfiled app demo Contract as Bucket Owner

### PR DESCRIPTION
**Pull request by cosinlink**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -223,6 +223,7 @@ nav:
                   - Resources Mirroring with CLI: ./bnb-greenfield/for-developers/tutorials/cross-chain-integration/mirror.md
                   - Integrating BSC Smart Contracts with Greenfield Projects: ./bnb-greenfield/for-developers/tutorials/cross-chain-integration/dapp-integration.md
                   - Mirroring FAQ: ./bnb-greenfield/for-developers/tutorials/cross-chain-integration/mirroring-faqs.md
+                  - Contract as Bucket Owner: ./bnb-greenfield/for-developers/tutorials/cross-chain-integration/demo-contract-as-bucket-owner.md
```
Comment (unclezoro) [inline mkdocs.yml:226]: I think the order is also important, `Mirroring FAQ` should always be the last one.

Comment (cosinlink) [reply mkdocs.yml:225]: fixed

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+```solidity
+    // BSC testnet
+    address public constant TOKEN_HUB = 0xED8e5C546F84442219A5a987EE1D820698528E04;
+    address public constant CROSS_CHAIN = 0xa5B2c9194131A4E0BFaCbF9E5D6722c873159cb7;
+    address public constant BUCKET_HUB = 0x5BB17A87D03620b313C39C24029C94cB5714814A;
+    address public constant PERMISSION_HUB = 0x25E1eeDb5CaBf288210B132321FBB2d90b4174ad;
+    address public constant SP_ADDRESS_TESTNET = 0x5FFf5A6c94b182fB965B40C7B9F30199b969eD2f;
+    address public constant GREENFIELD_EXECUTOR = 0x3E3180883308e8B4946C9a485F8d91F8b15dC48e;
+```
+The version of greenfield contracts deployed on BSC testnet is configured here.
```
Comment (KeefeL) [inline docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:25]: The addresses of Greenfield contracts deployed on the BSC testnet are available and configured here. You can find contracts deployed on other networks [here](***replace by the smart contract entrypoints directory here***)

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+```solidity
+    function createBucket(string memory bucketName, uint256 transferOutAmount, bytes memory _executorData) external payable;
+```
+Provides the bucket name, initial BNB amount transferred to the demo contract on Greenfield and the executor data 
+to set bucket flow rate limit.
+
+### create policy to allow eoa account to upload files to the bucket
+```solidity
+    function createPolicy(bytes memory createPolicyData) external payable;
+```
```
Comment (KeefeL) [inline docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:37]: Construct the `createPolicyData` parameters to grant permission to the dedicated principle for the created bucket owned by the smart contract.

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    address public constant GREENFIELD_EXECUTOR = 0x3E3180883308e8B4946C9a485F8d91F8b15dC48e;
+```
+The version of greenfield contracts deployed on BSC testnet is configured here.
+
+### create bucket and set bucket flow rate limit
+```solidity
+    function createBucket(string memory bucketName, uint256 transferOutAmount, bytes memory _executorData) external payable;
+```
+Provides the bucket name, initial BNB amount transferred to the demo contract on Greenfield and the executor data 
+to set bucket flow rate limit.
```
Comment (KeefeL) [inline docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:32]: Provides the bucket name, initial BNB amount transferred to the demo contract on Greenfield and the executor data 
to set bucket flow rate limit.This API will create a new bucket owned by the smart contract and set the payment for it.

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md (hunk 2 above)
Comment (cosinlink) [reply docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:25]: fixed

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md (hunk 3 above)
Comment (cosinlink) [reply docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:37]: fixed

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md (hunk 4 above)
Comment (cosinlink) [reply docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md:32]: fixed

**Review by KeefeL — APPROVED** [review]

**Review by unclezoro — APPROVED** [review]

**Comment by Bassambmbmbm** [comment]
مرحبا ... المعذرة
ممكن الخطوات

nu11

في الاثنين، ٨ يوليو ٢٠٢٤ ٥:٢١ ص zoro ***@***.***> كتب:

File: docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md (hunk 2 above)
Comment (Bassambmbmbm) [reply docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md]: ياشباب ...اخبروني  اضيف مستند العقد

**Pull request merged**